src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -119,16 +119,13 @@ const HeatmapPlot = (props) => {
     expressionValue]);
 
   useEffect(() => {
-    const louvainClusters = hierarchy.find((clusters) => clusters.key === 'louvain');
-
     if (louvainClustersResolution
-      && louvainClusters
-      && louvainClustersRef.current !== louvainClusters
+      && louvainClustersResolutionRef.current !== louvainClustersResolution
     ) {
-      louvainClustersRef.current = louvainClusters;
+      louvainClustersResolutionRef.current = louvainClustersResolution;
       dispatch(loadMarkerGenes(experimentId, louvainClustersResolution, COMPONENT_TYPE));
     }
-  }, [louvainClustersResolution, hierarchy]);
```

Comment:
Removed dependency on hierarchy which caused the double call, to check whether the clustering change we don't really need to depend on the actual louvain clusters, just the resolution so check on that